src/server/controllers/addNewUser.js

```diff
@@ -1,11 +1,23 @@
 const { createUser } = require('./../../database/db_create');
+const sendEmail = require('./../../helpers/emailVerification');
 
 module.exports = (req, res) => {
+  // create random 16 character token for email verification
+  const chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  let token = '';
+  for (let i = 16; i > 0; --i) {
+    token += chars[Math.round(Math.random() * (chars.length - 1))];
+  }
+
+  req.body.token = token;
+
   createUser(req.body, (error, response) => {
     if (error) {
       // will be refactored, so that err.message is displayed on screen
       res.status(400).send({ error: error.message });
     }
     res.redirect('/');
   });
+
+  sendEmail(req.body.username, req.body.email, token);
```

Comment:
Okay, right now you are responding in parallel to sending an email, if the email fails you have no way of communicating that with the client.

I would make send email behave like a normal async function, taking a callback function that it calls once it has finishing executing, passing an error if one has been raised. Even if right now you do not want to do anything with the error, you should structure your code so that the controller handles the error..